mailbox/event/event-memory/src/main/java/org/apache/james/mailbox/events/MemoryEventDeadLetters.java

```diff
@@ -83,4 +83,9 @@ public MemoryEventDeadLetters() {
             return Flux.fromIterable(ImmutableList.copyOf(deadLetters.rowKeySet()));
         }
     }
+
+    @Override
+    public Mono<Boolean> hasEvent() {
+        return Mono.just(deadLetters.size() > 0);
```

Comment:
What about `!deadLetters.isEmpty()` ?

Reply:
`!deadLetters.isEmpty()` still call `.size()` method and it not very impact to `readability` so i choose `.size()` method.

Reply:
Let me disagree: count + "greater than 0" is too much work compared to "empty", whatever the implementation

Reply:
BTW `!` and `deadLetters.isEmpty()` is 2 operations, same on `count + "greater than 0"`

Reply:
I'm not trying to count the number of symbols we write in Java. I'm talking about concepts: if I give you a possibly-empty box with objects in it and ask you to count the objects before telling me if it's empty or not, you will tell me that this is very inefficient because you have a faster way: if you see a single object, you can answer me right away.

So, counting when you have a `empty` method is always too much work.

It's called "Rule of least power": if two operations allow to achieve a given goal, always choose the least powerful.

Reply:
I agreed with **the rule**.